mesa/datacollection.py

```diff
@@ -175,8 +173,8 @@ def collect(self, model):
                 if isinstance(reporter, types.LambdaType):
                     self.model_vars[var].append(reporter(model))
                 # Check if model attribute
-                elif isinstance(reporter, partial):
-                    self.model_vars[var].append(reporter(model))
+                elif isinstance(reporter, str):
+                    self.model_vars[var].append(getattr(model, reporter, None))
```

Comment:
`self._getattr` is shorter.

Reply:
but it is an indirection we shouldn't make, calling a function which only calls a function is not useful, this is done only for calling partial because it swaps the parameters

Reply:
I thought the point of this PR is the speedup due to removing the partial. If you include removing `self._getattr`, then this could be the cause of the bulk of the speedup. It's unclear.

Reply:
indeed, I verified that a big part of the speed-up is due to the removal of `self._getattr` :-)

Reply:
Yes, the commit message & the PR title needs to be changed.

Reply:
Also, always prefix the perf commit message with "perf:". It's really hard to distinguish from a refactor otherwise.

Reply:
amended +1: